src/DependencyInjection/Core/Shared/DIToolkitPackage.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Threading;
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.Shell;
+using Task = System.Threading.Tasks.Task;
+using Community.VisualStudio.Toolkit.DependencyInjection.Core;
+
+namespace Community.VisualStudio.Toolkit.DependencyInjection
+{
+    /// <summary>
+    /// Package that contains a DI service container.
+    /// </summary>
+    /// <typeparam name="TPackage"></typeparam>
+    [ProvideService(typeof(SToolkitServiceProvider<>), IsAsyncQueryable = true)]
+    public abstract class DIToolkitPackage<TPackage> : ToolkitPackage
```

Comment:
Since this is `abstract`, do people like the name `Base` added to the end of this? Ie `DIToolkitPackageBase`